Modules/Core/TestKernel/include/itkTestingMacros.h

```diff
@@ -62,30 +62,45 @@ namespace itk
-    return EXIT_FAILURE;                                                                                               \
-  }                                                                                                                    \
-  if (!std::strcmp(object->Superclass::GetNameOfClass(), #SuperclassName))                                             \
-  {                                                                                                                    \
-    std::cout << "Superclass name is correct" << std::endl;                                                            \
-  }                                                                                                                    \
-  else                                                                                                                 \
-  {                                                                                                                    \
-    std::cerr << "Superclass name provided does not match object's Superclass::NameOfClass" << std::endl;              \
-    return EXIT_FAILURE;                                                                                               \
-  }                                                                                                                    \
-  ITK_MACROEND_NOOP_STATEMENT
+#  define ITK_EXERCISE_BASIC_OBJECT_METHODS(object, ClassName, SuperclassName)                                         \
+    object->Print(std::cout);                                                                                          \
+    std::cout << "Name of Class = " << object->GetNameOfClass() << std::endl;                                          \
+    if (!std::strcmp(object->GetNameOfClass(), #ClassName))                                                            \
+    {                                                                                                                  \
+      std::cout << "Class name is correct" << std::endl;                                                               \
+    }                                                                                                                  \
+    else                                                                                                               \
+    {                                                                                                                  \
+      std::cerr << "Class name provided does not match object's NameOfClass" << std::endl;                             \
+      return EXIT_FAILURE;                                                                                             \
+    }                                                                                                                  \
+    ITK_MACROEND_NOOP_STATEMENT
+#else // not GCC
+object->Print(std::cout);
```

Comment:
Should the body of the `else` statement be indented? Are the line continuation backslashes missing in the body of the `else` statement?

Reply:
> Although I do not have a better proposal (sorry), I am not sure avoiding to print the superclass name is what we would like as a mid-/long-term fix.

Ideally we would find the bug in GCC. However, the condition that causes this is not obvious and compilers are complicated. Letting false-positive failures clutter the dashboard prevents long term progress by masking other issues.

> I find particularly worrying that this is only happening for a few classes in the review module; the macro is used extensively throughout the code and the faulty statements are working well. So I think there is something more behind it that we are not being able to identify.

Yes, we should leave #2676 open, but that should not prevent this from cluttering the dashboard. The check is covered by Clang, Visual Studio. We are potentially missing other problems that GCC could bring up.

> As for the proposed fix, if this is what we aim at for now, am I wrong when saying that the only statements that should need to be within the if/else are those related to the superclass? 

Yes, the superclass statements are removed.

> That the rest of the code is exactly the same, and we could avoid duplicating it?

There is not a clean way to avoid duplicating it that I am aware of because of the use of macros inside macros.

Reply:
:+1: Sounds reasonable.

Reply:
>Should the body of the else statement be indented? Are the line continuation backslashes missing in the body of the else statement?

`clang-format` did a bit too much formatting :-)